parsers/IN_WE.py

```diff
@@ -29,48 +30,60 @@
 }
 
 KIND_MAPPING = {
-    "exchange": {"url": EXCHANGE_URL, "datetime_column": "lastUpdate"},
-    "consumption": {"url": CONSUMPTION_URL, "datetime_column": "current_datetime"},
+    "exchange": {
+        "url": EXCHANGE_URL,
+        "datetime_col": "lastUpdate",
+        "region_col": "Region_Name",
+    },
+    "consumption": {
+        "url": CONSUMPTION_URL,
+        "datetime_col": "current_datetime",
+        "region_col": "StateName",
+    },
 }
 
 
 def is_expected_downtime() -> bool:
-    current_day = datetime.now().weekday()
+    """
+    Details about this expected downtime is currently being clarified via
+    https://github.com/electricitymaps/electricitymaps-contrib/pull/6184#issuecomment-2564302911.
+    """
     expected_outage_days = [5, 6, 0]  # Saturday, Sunday and Monday
-
-    if current_day in expected_outage_days:
+    if datetime.now(ZONE_INFO).weekday() in expected_outage_days:
         return True
-
     return False
 
 
 def get_date_range(dt: datetime):
+    """Returns 24 datetime objects for a given datetime's date, one for each hour."""
+    now_date_as_dt = datetime.combine(
+        datetime.now(ZONE_INFO).date(), datetime.min.time()
+    ).replace(tzinfo=ZONE_INFO)
     return pd.date_range(
-        arrow.get(dt).floor("day").datetime,
-        arrow.get(dt).ceil("day").floor("hour").datetime,
+        now_date_as_dt,
+        now_date_as_dt + timedelta(hours=23),
         freq="H",
     ).to_pydatetime()
 
 
 def fetch_data(
-    kind: str | None = None,
-    session: Session | None = None,
-    target_datetime: datetime | None = None,
+    kind: str,
+    session: Session,
+    target_datetime: datetime,
 ) -> dict:
-    """- get production data from wrldc.in
-    - filter all rows with same hour as target_datetime"""
-    assert target_datetime is not None
-    assert kind is not None
+    """
+    Fetches 24 hours of either exchange or production data from wrldc.in.
+    """
+    url = KIND_MAPPING[kind]["url"]
+    datetime_col = KIND_MAPPING[kind]["datetime_col"]
 
-    r = session or Session()
     payload = {"date": target_datetime.strftime("%Y-%m-%d")}
-
-    resp: Response = r.post(url=KIND_MAPPING[kind]["url"], json=payload)
+    resp: Response = session.post(url, json=payload)
 
     try:
         data = json.loads(resp.json().get("d", {}))
     except Exception as e:
-        if is_expected_downtime():
+        if is_expected_downtime(target_datetime):
```

Comment:
This should not have a datetime passed as it cause type issues.

Reply:
Nice catch! I removed the `is_expected_downtime`, it seems to work but its currently such downtime, so I think they have made it work reliably.